File: backend-dotnet/SahayaSetu/Controllers/AdminController.cs
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using SahayaSetu.Services;

namespace SahayaSetu.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class AdminController : ControllerBase
    {
        private readonly IAdminService _adminService;

        public AdminController(IAdminService adminService)
        {
            _adminService = adminService;
        }

        [HttpGet("ngos")]
        public async Task<IActionResult> GetAllNgos()
        {
            var result = await _adminService.GetAllNgosAsync();
            return Ok(result);
        }

        [HttpGet("donors")]
        public async Task<IActionResult> GetAllDonors()
        {
            var result = await _adminService.GetAllDonorsAsync();
            return Ok(result);
        }
        
        [HttpGet("requests")]
        public async Task<IActionResult> GetAllRequests()
        {
            var result = await _adminService.GetAllRequestsAsync();
            return Ok(result);
        }

        [HttpPut("ngos/{ngoId}/approve")]
        public async Task<IActionResult> ApproveNgo(int ngoId)
        {
            await _adminService.ApproveNgoAsync(ngoId);
            return Ok("NGO approved successfully");
        }

        [HttpPut("ngos/{ngoId}/disable")]
        public async Task<IActionResult> DisableNgo(int ngoId)
        {
            await _adminService.DisableNgoAsync(ngoId);
            return Ok("NGO disabled successfully");
        }
        
        [HttpPut("ngos/{ngoId}/disapprove")]
        public async Task<IActionResult> DisapproveNgo(int ngoId)
        {
            await _adminService.DisapproveNgoAsync(ngoId);
            return Ok("NGO disapproved successfully");
        }
        
        [HttpPut("ngos/{ngoId}/enable")]
        public async Task<IActionResult> EnableNgo(int ngoId)
        {
            await _adminService.EnableNgoAsync(ngoId);
            return Ok("NGO enabled successfully");
        }

    }
}

File: backend-java/sahayaSetu/src/main/java/com/sahayaSetu/controllers/AdminController.java
package com.sahayaSetu.controllers;

import com.sahayaSetu.services.IAdminService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;

import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/admin")
@RequiredArgsConstructor
public class AdminController {

    private final IAdminService adminService;

    @GetMapping("/ngos")
    public ResponseEntity<?> getAllNgos() {
        return ResponseEntity.ok(adminService.getAllNgos());
    }

    @GetMapping("/requests")
    public ResponseEntity<?> getAllRequests() {
        return ResponseEntity.ok(adminService.getAllRequests());
    }

    @GetMapping("/donors")
    public ResponseEntity<?> getAllDonors() {
        return ResponseEntity.ok(adminService.getAllDonors());
    }

    @PutMapping("/ngos/{id}/approve")
    public ResponseEntity<?> approveNgo(@PathVariable Long id) {
        adminService.approveNgo(id);
        return ResponseEntity.ok("NGO approved successfully");
    }

    @PutMapping("/ngos/{id}/disapprove")
    public ResponseEntity<?> disapproveNgo(@PathVariable Long id) {
        adminService.disapproveNgo(id);
        return ResponseEntity.ok("NGO disapproved successfully");
    }

    @PutMapping("/ngos/{id}/disable")
    public ResponseEntity<?> disableNgo(@PathVariable Long id) {
        adminService.disableNgo(id);
        return ResponseEntity.ok("NGO disabled successfully");
    }

    @PutMapping("/ngos/{id}/enable")
    public ResponseEntity<?> enableNgo(@PathVariable Long id) {
        adminService.enableNgo(id);
        return ResponseEntity.ok("NGO enabled successfully");
    }
}
